py/desispec/fiberflat_vs_humidity.py

```diff
@@ -0,0 +1,128 @@
+"""
+desispec.fiberflat_vs_humidity
+==================
+
+Utility functions to compute a fiber flat corrected for variations with humidity in the shack
+"""
+from __future__ import absolute_import, division
+
+
+import numpy as np
+import copy
+
+from desiutil.log import get_logger
+from desispec.fiberflat import apply_fiberflat
+from desispec.fiberbitmasking import get_skysub_fiberbitmask_val
+
+def _interpolated_fiberflat_vs_humidity(fiberflat_vs_humidity , humidity_array, humidity_point) :
+
+    if humidity_point<=humidity_array[0] :
+        i1=0
+    else :
+        i1=np.where(humidity_array<humidity_point)[0][-1]
+    i2=i1+1
+    if i2>=humidity_array.size : # return largest value
+        return fiberflat_vs_humidity[-1]
+
+    w1=(humidity_array[i2]-humidity_point)/(humidity_array[i2]-humidity_array[i1])
+    w2=(humidity_point-humidity_array[i1])/(humidity_array[i2]-humidity_array[i1])
+    return w1*fiberflat_vs_humidity[i1]+w2*fiberflat_vs_humidity[i2]
+
+def _fit_flat(wavelength,flux,ivar,fibers,mean_fiberflat_vs_humidity,humidity_array) :
+
+    log = get_logger()
+    selection = (wavelength > 4000.) & (wavelength < 4600)
+    if np.sum(selection)==0 :
+        message="incorrect wavelength range"
+        log.error(message)
+        raise RuntimeError(message)
+    waveindex = np.where(selection)[0]
+    tmp_flux = flux[fibers][:,waveindex].copy()
+    tmp_ivar = ivar[fibers][:,waveindex].copy()
+
+    for loop in range(2) :
+        # remove mean variation from fiber to fiber
+        med   = np.median(tmp_flux,axis=-1)
+        tmp_flux /= med[:,None]
+        tmp_ivar *= med[:,None]**2
+        # remove average over fibers
+        med =  np.median(tmp_flux,axis=0)
+        tmp_flux /= med[None,:]
+        tmp_ivar *= med[None,:]**2
+
+    tmp_flat = mean_fiberflat_vs_humidity[:,fibers][:,:,waveindex].copy()
+
+    for loop in range(2) :
+        # remove mean variation from fiber to fiber
+        med   = np.median(tmp_flat,axis=-1)
+        tmp_flat /= med[:,:,None]
+        # remove average over fibers
+        for index in range(tmp_flat.shape[0]) :
+            med =  np.median(tmp_flat[index],axis=0)
+            tmp_flat[index] /= med[None,:]
+
+    chi2 = np.sum(np.sum(tmp_ivar*(tmp_flux-tmp_flat)**2,axis=-1),axis=-1)
```

Comment:
I think I know what you are doing here, but please add some comments for our future selves about what this fit is doing and why.

Reply:
done